enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,641 @@
+
+[Cluster-API Provider Bare Metal](https://github.com/openshift/cluster-api-provider-baremetal/) will need to gain the ability to interact with the new assisted installer APIs. Specifically it will need to match a Machine with an available assisted Agent that is ready to be provisioned. This will be in addition to matching a Machine with a BareMetalHost, which it already does today.
+
+#### Day 2 Add Node Boot-it-Yourself
+
+This scenario takes place within a stand-alone bare metal OpenShift cluster.
+
+1. The user downloads a discovery ISO from the cluster. The download is implemented by the assisted installer as a URL on a AssistedInstallCluster
+resource.
+1. A host boots the live ISO, the assisted agent starts running, and the agent contacts the assisted service to register its existence. Communication utilizes the existing non-k8s REST API. The agent walks through the validation and inspection workflow as it exists today.
+1. The wrapping controller creates a new AssistedInstallAgent resource to be the k8s-native API for the agent.
+1. CAPBM creates a new BareMetalHost resource, setting the status annotation based on inspection information from the prior step.
+1. The user approves the host for installation. (TODO: how?)
+1. The user or an orchestrator scales up a MachineSet, causing a new Machine resource to be created.
+1. CAPBM binds the Machine to the BareMetalHost, as it does today. It additionally finds the CR representing the agent and uses it to begin installation.
+1. The assisted service initiates installation of the host.
+1. CAPBM updates the status on the BareMetalHost to reflect that it has been provisioned.
+
+#### Day 2 Add Node Virtualmedia Stand-alone
+
+This scenario takes place within a stand-alone bare metal OpenShift cluster.
+
+1. The user creates a BareMetalHost that includes BMC credentials and a label indicating it should be used with assisted installer.
+1. CAPBM gets a URL to the live ISO and adds it to the BareMetalHost, causing it to boot the host.
+1. baremetal-operator uses redfish virtualmedia to boot the live ISO.
+1. The assisted agent starts running on the new hardware and runs through its usual validation and inspection workflow. The wrapping controller creates a new agent resource to be the k8s-native API for the agent.
+1. The user or an orchestrator scales up a MachineSet, resulting in a new Machine being created.
+1. CAPBM does its usual workflow of matching the Machine to an available BareMetalHost. Additionally it uses the agent CR to initiate provisioning of the host.
+1. The assisted service provisions the host.
+
+#### Day 2 Add Node Virtualmedia Multicluster
+
+This scenario takes place from a hub cluster, adding a worker node to a spoke cluster.
+
+1. User creates or modifies a BareMetalAsset on the hub cluster so that its cluster field references the desired ClusterDeployment.
+1. Hive creates a BareMetalHost resource corresponding to the BareMetalAsset and uses it to boot the live ISO.
```

Comment:
We created separate Asset and Host resource types in the past because it was possible the hub cluster was not going to have access to the BMCs and we didn't want the full Host type to be used just for inventory management. In this case, the hub cluster *must* have access to the BMC, so why not just have the user create the Host directly?